components/mpas-ocean/src/shared/mpas_ocn_scaled_sfwf.F

```diff
@@ -0,0 +1,557 @@
+
+        ! Setup Alarm for updating of scaling factors for precipitation flux
+        alarmTime = mpas_get_clock_time(domain % clock, &
+                                        mpas_START_TIME, &
+                                        ierr=err_tmp)
+        
+        call mpas_set_timeInterval(alarmTimeStep, &
+                                   timeString=timeString,& 
+                                   ierr=err_tmp)
+
+        call mpas_add_clock_alarm(domain % clock, alarmID, alarmTime + alarmTimeStep, &
+                                  alarmTimeInterval=alarmTimeStep, ierr=err_tmp)
+                          
+        ! Compute Earth's surface area (used for normalizing fluxes)
+        earthAreaE3SM = 4.0_RKIND * piE3SM * earthRadiusE3SM*earthRadiusE3SM
+
+        ! Density of sea water
+        rhow = rho_sw
+
+    end subroutine ocn_init_scaled_sfwf!}}}
+
+
+!***********************************************************************
+!
+!  routine ocn_update_scaling_factor
+!
+!> \brief  Update scaling factors for data freshwater fluxes
+!> \author Shixuan Zhang 
+!> \date   May 2025
+!> \details
+!>  Accumulate yearly mean of surface freshwater fluxes. 
+!>  If we are at the end of a year, used it to derive a factor to scale 
+!>  down/up precipitation fluxes and maintain freshwater balance.  
+!
+!-----------------------------------------------------------------------
+
+    subroutine ocn_update_scaling_factor(domain,timeLevel)!{{{
+
+       type (domain_type), intent(inout) :: domain
+       
+       integer, intent(in), optional :: timeLevel
+
+       integer :: timeLevelLocal
+       
+       integer :: err_tmp
+
+       if (.not. scaledSFWFOn) then
+           return
+       end if
+
+       if ( present(timeLevel) ) then
+          timeLevelLocal = timeLevel
+       else
+          timeLevelLocal = 1
+       end if
+
+       ! since the current clock time is for the end of the accumulation
+       ! intereval, first, accumulate the yearly mean
+       call accumulate_mean_fw_fluxes(domain,timeLevelLocal)
+
+       ! then, compute update the history and running mean if we are at the
+       ! end of a year
+       if(mpas_is_alarm_ringing(domain % clock, alarmID, ierr=err_tmp)) then
+#ifdef mpas_DEBUG
+            call mpas_log_write('       Computing Scaling Factor For Water Fluxes')
+#endif
+            call update_scaling_factor(domain,timeLevelLocal)
+            call mpas_reset_clock_alarm(domain % clock, alarmID, ierr=err_tmp)
+       endif
+
+    end subroutine ocn_update_scaling_factor!}}}
+
+!***********************************************************************
+!
+!  routine compute_ssh_mass
+!
+!> \brief  Derive ssh inferred freshwater mass 
+!> \author Shixuan Zhang
+!> \date   May 2025
+!> \details
+!>  calculate ssh equivalent total mass over ocean grid point 
+!
+!-----------------------------------------------------------------------
+    subroutine compute_ssh_mass(domain,timeLevel,sshMass)!{{{
```

Comment:
```suggestion
    subroutine ocn_compute_mass_from_ssh(domain,timeLevel,sshMass)!{{{
```

Reply:
The `ocn_` prefix is not needed for subroutines that are private, and omitting it is often a useful indication that a subroutine is not public.